hugr/src/extension/op_def.rs

```diff
@@ -662,4 +731,82 @@ mod test {
         );
         Ok(())
     }
+
+    #[cfg(feature = "proptest")]
+    mod proptest {
+        use super::SimpleOpDef;
+        use ::proptest::prelude::*;
+
+        use crate::{
+            builder::test::simple_dfg_hugr,
+            extension::{
+                op_def::LowerFunc, CustomValidator, ExtensionId, ExtensionSet, OpDef, SignatureFunc,
+            },
+            types::PolyFuncType,
+        };
+
+        impl Arbitrary for SignatureFunc {
+            type Parameters = ();
+            type Strategy = BoxedStrategy<Self>;
+            fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+                // TODO there is also  SignatureFunc::CustomFunc, but for now
```

Comment:
Which means that if the recipient has the same struct+trait code available, he'll get back an instance of the same struct, and this'll work fine (and since for a roundtrip the "recipient" *is* the "sender", that'll be fine). Of course that can't work if the recipient doesn't have the code, but nvm, that's not really relevant here :)